java/src/jmri/jmrit/DccLocoAddressSelector.java

```diff
@@ -190,28 +193,32 @@ private void changeFontSizes() {
             return;
         }
         double fieldWidth = text.getSize().width;
-        int stringWidth = text.getFontMetrics(text.getFont()).
-                stringWidth(LONGEST_STRING) + 8;
+        int stringWidth = text.getFontMetrics(text.getFont()).stringWidth(LONGEST_STRING) + 8;
         int fontSize = text.getFont().getSize();
-        if (stringWidth > fieldWidth) // component has shrunk.
-        {
-            while ((stringWidth > fieldWidth) && (fontSize > 12)) {
+        if (stringWidth > fieldWidth) { // component has shrunk horizontally
+            while ((stringWidth > fieldWidth) && (fontSize >= FONT_SIZE_MIN)) {
                 fontSize -= 2;
```

Comment:
just did, using strict inequality I don't know why I introduced these non strict inequality, thanks.

Reply:
It would be better if you kept non strict inequality and used `(fontSize >= FONT_SIZE_MIN + FONT_INCREMENT)`.

The original code might have a bug in it, depending on the intent of the original coder. Let's assume that the font size is 13 and the min font size is 12. Then with strict inequality, as it was in the beginning, the font size would be decreased with 2, and the resulting font size would be 11.

With the code I suggested, the font size will be guaranteed to be greater or equal to the minimum font size. And it would still work if the `FONT_INCREMENT` is increased to 4 or 6.

(Screens are getting bigger and bigger. When I started working with computers 30 years ago, 640x480 pixels was high resolution. Now people are using screens with 4096 x 2160 pixels resolution).

Reply:
done.

(Regarding screens, it took 34 years to move from my Atari ST 320x200 to to the incredible 5120x1440 Philips screen I'm using now, which looks a lot, but is actually way less than the increase in CPU and memory performances.)

Reply:
Thanks!